src/main/java/seedu/address/logic/commands/DeleteAttendanceCommand.java

```diff
@@ -0,0 +1,167 @@
+            + "Parameters: "
+            + PREFIX_CLASS_INDEX + "CLASS_INDEX "
+            + PREFIX_LESSON_INDEX + "LESSON_INDEX "
+            + PREFIX_STUDENT_INDEX + "STUDENT_INDEX "
+            + PREFIX_WEEK + "WEEK_INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_CLASS_INDEX + "1 "
+            + PREFIX_LESSON_INDEX + "1 "
+            + PREFIX_STUDENT_INDEX + "1 "
+            + PREFIX_WEEK + "1";
+
+    public static final String MESSAGE_DELETE_ATTENDANCE_SUCCESS =
+            "Deleted week %1$s attendance of student %2$s from lesson %3$s";
+
+    private final Index moduleClassIndex;
+    private final Index lessonIndex;
+    private final Index studentIndex;
+    private final Week week;
+
+    /**
+     * @param moduleClassIndex in the filtered class list.
+     * @param lessonIndex in the filtered lesson list.
+     * @param studentIndex in the filtered student list.
+     * @param week in the specified attendance list to be deleted.
+     */
+    public DeleteAttendanceCommand(
+            Index moduleClassIndex, Index lessonIndex, Index studentIndex, Week week) {
+        requireAllNonNull(moduleClassIndex, lessonIndex, studentIndex, week);
+
+        this.moduleClassIndex = moduleClassIndex;
+        this.lessonIndex = lessonIndex;
+        this.studentIndex = studentIndex;
+        this.week = week;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Student> lastShownStudentList = model.getFilteredStudentList();
+        List<ModuleClass> lastShownModuleClassList = model.getFilteredModuleClassList();
+
+        if (studentIndex.getZeroBased() >= lastShownStudentList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+        }
+
+        if (moduleClassIndex.getZeroBased() >= lastShownModuleClassList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_CLASS_DISPLAYED_INDEX);
+        }
+
+        Student targetStudent = lastShownStudentList.get(studentIndex.getZeroBased());
+        ModuleClass targetModuleClass = lastShownModuleClassList.get(moduleClassIndex.getZeroBased());
+
+        if (lessonIndex.getZeroBased() >= targetModuleClass.getLessons().size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_LESSON_DISPLAYED_INDEX);
+        }
+
+        Lesson targetLesson = targetModuleClass.getLessons().get(lessonIndex.getZeroBased());
+
+        if (!targetLesson.getAttendanceRecordList().isWeekContained(week)) {
+            throw new CommandException(Messages.MESSAGE_INVALID_WEEK);
+        }
+
+        Lesson modifiedLesson = createModifiedLesson(targetLesson, targetStudent, week);
+        ModuleClass modifiedModuleClass = createModifiedModuleClass(targetModuleClass, lessonIndex, modifiedLesson);
+        model.setModuleClass(targetModuleClass, modifiedModuleClass);
```

Comment:
I am wondering if these lines, along with the private methods declared below, can be abstracted into a class that deals with the manipulation of the `Lesson` and `Attendance` related classes. These lines of code are repeated in other commands and in test cases as well.

Reply:
Created ```ModuleClassModificationUtil``` and ```LessonModificationUtil``` classes to address this